### PR TITLE
Add the TARGET_LIBS environment variable for rustc CI testing

### PR DESCRIPTION
**Pull request by msizanoen1**
Needed to fix the test failure in rust-lang/rust#66158.

See https://github.com/rust-lang/rust/pull/66158#issuecomment-550585396

r? @Manishearth

changelog: none

### CODE REVIEW

**Comment by lzutao** [comment]
Ping @flip1995 @msizanoen1 
@msizanoen1 you might need to rebase again upstream master.

File: tests/compile-test.rs [hunk 1]
```diff
@@ -58,7 +63,14 @@ fn config(mode: &str, dir: PathBuf) -> compiletest::Config {
     let needs_disambiguation = ["serde", "regex", "clippy_lints"];
     // This assumes that deps are compiled (they are for Cargo integration tests).
     let deps = std::fs::read_dir(host_libs().join("deps")).unwrap();
+    let deps: Vec<_> = if let Some(target_libs) = target_libs() {
+        deps.chain(std::fs::read_dir(target_libs.join("deps")).unwrap())
```
Comment (lzutao) [inline tests/compile-test.rs]:
```suggestion
        deps.chain(fs::read_dir(target_libs.join("deps")).unwrap())
```

File: tests/compile-test.rs [hunk 2]
```diff
@@ -58,7 +63,14 @@ fn config(mode: &str, dir: PathBuf) -> compiletest::Config {
     let needs_disambiguation = ["serde", "regex", "clippy_lints"];
     // This assumes that deps are compiled (they are for Cargo integration tests).
     let deps = std::fs::read_dir(host_libs().join("deps")).unwrap();
```
Comment (lzutao) [inline tests/compile-test.rs]:
```suggestion
     let deps = fs::read_dir(host_libs().join("deps")).unwrap();
```

Comment (lzutao) [reply tests/compile-test.rs]: ?

Comment (msizanoen1) [reply tests/compile-test.rs]: I forgot to `git add`.

**Comment by lzutao** [comment]
build failed because of format check. Could you run `./util/dev fmt`?

**Comment by Manishearth** [comment]
Do you think this will work when cross compiling? I'm strongly inclined to just disable these tests when running rustc ci.

**Comment by mati865** [comment]
Tools tests aren't ran when cross compiling right now.
In fact they are ran only on `x86_64-gnu-tools` and `x86_64-msvc-tools` which are native but user could enable them for local build.

**Comment by Manishearth** [comment]
Oh, right!

Anyway, I found a better fix for this in #4784, we really shouldn't be linking to clippy_lints in those tests anyway.

**Comment by msizanoen1** [comment]
Actually even with that PR you will still get the `regex` and `serde` crate not found error.

**Comment by msizanoen1** [comment]
About the tools are tested only when building natively assumption, the rustbuild build system always pass the `--target` flags so the target rlibs are put in a separate directory.

**Comment by msizanoen1** [comment]
Cross compilation and testing is not a goal of this PR.

File: tests/compile-test.rs [hunk 3]
```diff
@@ -27,6 +27,11 @@ fn host_libs() -> PathBuf {
     }
 }
 
+#[must_use]
```
Comment (Manishearth) [inline tests/compile-test.rs]: the must_use isn't necessary

**Comment by Manishearth** [comment]
@brors r+

will land for now and maybe make minor fixes later

**Comment by msizanoen1** [comment]
> @Brors r+
Who's `Brors`?

**Comment by flip1995** [comment]
@bors retry

Changelog was missing

**Pull request merged**